src/federation/observer.rs

```diff
@@ -77,23 +77,36 @@ impl FederationObserver {
     }
 
     async fn setup_schema(&self) -> anyhow::Result<()> {
-        let schema_version = query_value::<i32>(
+        execute(
             &self.connection().await?,
-            // language=postgresql
-            "SELECT
-                    CASE
-                        WHEN EXISTS (
-                            SELECT FROM information_schema.tables
-                            WHERE table_schema = 'public'
-                            AND table_name = 'schema_version'
-                        )
-                        THEN (SELECT COALESCE(MAX(version), 0) FROM schema_version)
```

Comment:
I'm getting the following error when running the migrations. It looks like postgres still runs subqueries in the `THEN` branch even if `WHEN EXISTS` is false. Creating a function to handle works on my end.

```
Error: db error: ERROR: relation "schema_version" does not exist

Caused by:
    ERROR: relation "schema_version" does not exist
```

Reply:
Tested your version and it works